media/base/media_switches.h

```diff
@@ -99,6 +99,10 @@ MEDIA_EXPORT extern const char kLacrosUseChromeosProtectedMedia[];
 MEDIA_EXPORT extern const char kLacrosUseChromeosProtectedAv1[];
 #endif  // BUILDFLAG(IS_CHROMEOS)
 
+#if BUILDFLAG(USE_STARBOARD_MEDIA)
+MEDIA_EXPORT extern const char kDisableStarboardRenderer[];
```

Comment:
I think we should use a `BASE_DECLARE_FEATURE` [1] ISO
this "classic" command line switch.

[1] https://source.chromium.org/chromium/chromium/src/+/main:base/feature_list.h;drc=3604009eb6f4c2287d0052ed3eaaa62bff79541d;l=64

Reply:
I think this is media feature, not base feature?

Reply:
`BASE_DECLARE_FEATURE` is the generic way to declare base::Features, and is used in this same file a bit further below (e.g. [here](https://source.chromium.org/chromium/chromium/src/+/main:media/base/media_switches.h;l=174;drc=86b96ca15e8aab43465dbbcc30294e72ef0daafe)). 

The current syntax declares a [switch](https://chromium.googlesource.com/chromium/src/+/main/docs/configuration.md#switches) and in general a [feature](https://chromium.googlesource.com/chromium/src/+/main/docs/configuration.md#flags) is preferred.

Reply:
I found two issues if we use it as a feature instead of a switch:
1. Feature prefer enable define, which make this as `kEnableStarboardRenderer`, but we generally want it to be enabled by default, and only disable it if we want to.
2. `CobaltActivity.java` needs this to determine if it needs to create SurfaceView for SbPlayer, if we define in this way, then we need to add arguments like `--enable-starboard-renderer=true` or `--enable-starboard-renderer=false` to disable it to command line. This cannot be pre-added like https://github.com/youtube/cobalt/pull/4329, because we want to keep enabling or disabling it at the run time, which makes launching Cobalt with this as a necessary command line arguments very annoying.

I actually refer other media switches design, like [kForceVideoOverlays](https://source.chromium.org/chromium/chromium/src/+/main:media/base/media_switches.cc;bpv=1;bpt=1;l=179?gsn=kForceVideoOverlays&gs=KYTHE%3A%2F%2Fkythe%3A%2F%2Fchromium.googlesource.com%2Fcodesearch%2Fchromium%2Fsrc%2F%2Fmain%3Flang%3Dc%252B%252B%3Fpath%3Dmedia%2Fbase%2Fmedia_switches.cc%23qC1CuAeGL1l4bM0MtrEBh1xfZeYSjOBt20zON2Prr9o), which seems make the use case easier.

What do you think?

Reply:
base::Features can also be enabled or disabled by default ([example](https://source.chromium.org/chromium/chromium/src/+/main:media/base/media_switches.cc;l=1564-1571;drc=3604009eb6f4c2287d0052ed3eaaa62bff79541d)) -- you could replicate your scenario of `StarboardRenderer` by setting it as enabled by default, and then disable it via the command line `--disable-features=StarboardRenderer`. But since this seems to be a debug/developer flag (i.e. it won't end up in chrome:flags, nor be tracked via UMA), I guess it's fine like this.